ci/build_wheel.sh

```diff
@@ -7,6 +7,25 @@ package_name=$1
 package_dir=$2
 package_type=$3
 
+# The 'libcuspatial' wheel should package 'libcuspatial.so', and all others
+# should exclude it (they dynamically load it if they need it).
+#
+# Capturing that here in argument-parsing to allow this build_wheel.sh
+# script to be re-used by all wheel builds in the project.
+case "${package_dir}" in
+  python/libcuspatial)
+    EXCLUDE_ARGS=(
+      --exclude "libcudf.so"
```

Comment:
Just making sure — there are no other libraries to exclude like nvcomp?

Reply:
Thanks for checking! I don't think so.

Here are the libraries I see in the `libcuspatial` wheels from this branch:

```text
libcuspatial/lib64/libcuspatial.so
libcuspatial_cu12.libs/libcudart-16e8815a.so.12.5.82
```

And `cuspatial`

```text
cuspatial/_lib/quadtree.cpython-311-x86_64-linux-gnu.so
cuspatial/_lib/intersection.cpython-311-x86_64-linux-gnu.so
cuspatial/_lib/utils.cpython-311-x86_64-linux-gnu.so
cuspatial/_lib/polygon_bounding_boxes.cpython-311-x86_64-linux-gnu.so
cuspatial/_lib/points_in_range.cpython-311-x86_64-linux-gnu.so
cuspatial/_lib/spatial.cpython-311-x86_64-linux-gnu.so
cuspatial/_lib/distance.cpython-311-x86_64-linux-gnu.so
cuspatial/_lib/point_in_polygon.cpython-311-x86_64-linux-gnu.so
cuspatial/_lib/nearest_points.cpython-311-x86_64-linux-gnu.so
cuspatial/_lib/linestring_bounding_boxes.cpython-311-x86_64-linux-gnu.so
cuspatial/_lib/pairwise_point_in_polygon.cpython-311-x86_64-linux-gnu.so
cuspatial/_lib/trajectory.cpython-311-x86_64-linux-gnu.so
cuspatial/_lib/pairwise_multipoint_equals_count.cpython-311-x86_64-linux-gnu.so
cuspatial/_lib/spatial_join.cpython-311-x86_64-linux-gnu.so
cuspatial/_lib/types.cpython-311-x86_64-linux-gnu.so
```

Reply:
Looks good to me!

Reply:
Oh actually, maybe not. (Missed this on first look.) We should be statically linking libcudart, I think — or we should have a CUDA wheel dependency if we’re not statically linking it. Can you check what libcudf wheels are doing?

Reply:
`libcudf` appears to be statically linking `libcudart`.

Looked at the libraries in a recent `libcudf` nightly wheel

```shell
pip download --no-deps 'libcudf-cu12==24.10.*,>=0.0.0a0'
unzip -l ./libcudf_cu12-24.10.0a283-py3-none-manylinux_2_28_x86_64.whl | grep -E '\.a|\.so
```

No `libcudart`:

```text
libcudf/lib/libnanoarrow.a
libcudf/lib64/libnvcomp_device.a
libcudf/lib64/libnvcomp_gdeflate.so
libcudf/lib64/libnvcomp.so
libcudf/lib64/libnvcomp_gdeflate_cpu.so
libcudf/lib64/libflatbuffers.a
libcudf/lib64/libnvcomp_bitcomp.so
libcudf/lib64/libcudf.so
```

And in `libcudf`'s wheel CMake:

```cmake
set(CUDA_STATIC_RUNTIME ON)
```

([code link](https://github.com/rapidsai/cudf/blob/branch-24.10/python/libcudf/CMakeLists.txt#L42))

I can attempt a similar pattern here in `libcuspatial`.

Reply:
Just pushed https://github.com/rapidsai/cuspatial/pull/1450/commits/1d00c91f4e72a0984883d403f13753e22750c86e, changing `libcuspatial` to statically linking `libcudart`.

This is how `cuspatial` wheels worked before this PR:

https://github.com/rapidsai/cuspatial/blob/1544e7bd16f11457436d1fb6a43da0db3aa9af57/python/cuspatial/CMakeLists.txt#L39

So it was a mistake to switch that to dynamic linking.

I updated the size table in the PR description... this change cut 1.0MB out of the `libcuspatial` wheel.

Also checked the wheel contents again...

libraries in `libcuspatial`:

```text
libcuspatial/lib64/libcuspatial.so
```

libraries in `cuspatial`:

```text
cuspatial/_lib/quadtree.cpython-311-x86_64-linux-gnu.so
cuspatial/_lib/intersection.cpython-311-x86_64-linux-gnu.so
cuspatial/_lib/utils.cpython-311-x86_64-linux-gnu.so
cuspatial/_lib/polygon_bounding_boxes.cpython-311-x86_64-linux-gnu.so
cuspatial/_lib/points_in_range.cpython-311-x86_64-linux-gnu.so
cuspatial/_lib/spatial.cpython-311-x86_64-linux-gnu.so
cuspatial/_lib/distance.cpython-311-x86_64-linux-gnu.so
cuspatial/_lib/point_in_polygon.cpython-311-x86_64-linux-gnu.so
cuspatial/_lib/nearest_points.cpython-311-x86_64-linux-gnu.so
cuspatial/_lib/linestring_bounding_boxes.cpython-311-x86_64-linux-gnu.so
cuspatial/_lib/pairwise_point_in_polygon.cpython-311-x86_64-linux-gnu.so
cuspatial/_lib/trajectory.cpython-311-x86_64-linux-gnu.so
cuspatial/_lib/pairwise_multipoint_equals_count.cpython-311-x86_64-linux-gnu.so
cuspatial/_lib/spatial_join.cpython-311-x86_64-linux-gnu.so
cuspatial/_lib/types.cpython-311-x86_64-linux-gnu.so
```

Thanks for catching that @bdice